src/projects/dualBackedYieldToOne/MDualBackedYieldToOne.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: BUSL-1.1
+
+pragma solidity 0.8.26;
+
+import { console } from "forge-std/console.sol";
+
+import { IERC20 } from "../../../lib/common/src/interfaces/IERC20.sol";
+
+import { MYieldToOne } from "../yieldToOne/MYieldToOne.sol";
+
+import { IMYieldToOne } from "../yieldToOne/IMYieldToOne.sol";
+
+import { IMDualBackedYieldToOne } from "./IMDualBackedYieldToOne.sol";
+
+import { ISwapFacility } from "../../swap/interfaces/ISwapFacility.sol";
+
+import { IMTokenLike } from "../../interfaces/IMTokenLike.sol";
+
+abstract contract MDualBackedToYieldOneStorageLayout {
+    struct MDualBackedYieldToOneStorageStruct {
+        address secondaryBacker;
+        uint256 secondarySupply;
```

Comment:
consider checking the actual balance of USDC/secondary backing supply vs tracking it

Reply:
If we were to check the actual balance of the secondary token, that could throw a wrench into how the contract functions. Although, it'd be an expensive mistake for someone to make, if the secondary were somehow sent directly to the contract without minting any tokens for it, it would confuse the contract and possibly prevent legitimate actions of claiming yield or unwrapping

also it could discombobulate how the extension replaces the secondary asset, it could wind up with more M behind it than it has minted tokens to represent.

Reply:
On second thought, claiming yield in such a scenario that would have the effect of crediting the USDC balance to the yield recipient - not creating false yield.

Reply:
I've removed it in this commit: https://github.com/m0-foundation/evm-m-extensions/pull/54/commits/e1e009de9119eb43a1cb04700ef4b2a528dd31b5
We can just check the `balanceOf` USDC: https://github.com/m0-foundation/evm-m-extensions/blob/e1e009de9119eb43a1cb04700ef4b2a528dd31b5/src/projects/dualBackedYieldToOne/MDualBackedYieldToOne.sol#L250